crates/core/tedge_mapper/src/c8y/converter.rs

```diff
@@ -834,11 +875,33 @@ pub fn get_child_id_from_topic(topic: &str) -> Result<Option<String>, Conversion
     }
 }
 
+pub fn get_child_id_from_event_topic(topic: &str) -> Result<Option<String>, ConversionError> {
+    match topic.strip_prefix("tedge/events/").map(String::from) {
+        Some(child_ev_topic) => {
+            let v: Vec<&str> = child_ev_topic.splitn(2, '/').collect();
+            if v.len() >= 2 {
```

Comment:
`v.len() >=2` means that this condition accepts `tedge/events/event_type/child_id/and_so_on`. It's an error.

Reply:
Removed this function, refactored the code.